### PR TITLE
Image tags

### LINKED ISSUE #7: Image Tags
#### As a user I want to add tags to images.

Acceptance criteria:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image on the index page.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

Reference material:
- https://github.com/mbleigh/acts-as-taggable-on

Dependencies:
- Image Index

### PR DESCRIPTION
**Pull request by tlconnor**
Closes #7

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class ImagesController < ApplicationController
   def index
-    @images = Image.all
+    @images = Image.includes(:tags)
```
Comment (bboe) [inline app/controllers/images_controller.rb]: Nice.

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def show
 
   def create
     @image = Image.new(params.permit(:url))
+    @image.tag_list =  params[:tags].join(', ') if params[:tags]
```
Comment (bboe) [inline app/controllers/images_controller.rb]: This line should be able to be combined with the above line.

File: db/schema.rb [hunk 3]
```diff
@@ -21,4 +21,29 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "taggings", force: :cascade do |t|
```
Comment (bboe) [inline db/schema.rb]: Do all of these fields require being nullable? Same with the "tags" table.

File: test/models/image_test.rb [hunk 4]
```diff
@@ -17,4 +17,10 @@ def test_invalid_image_url
     assert_predicate image, :invalid?
     assert_equal ['is not a valid image URL'], image.errors[:url]
   end
+
+  def test_image_tags
+    image = Image.new url: 'invalid value'
+    image.tag_list = 'a, b, c, d'
+    puts image.tags
```
Comment (bboe) [inline test/models/image_test.rb]: What does this test?

**Comment by bboe** [comment]
Added some comments.

File: db/schema.rb (hunk 3 above)
Comment (tlconnor) [inline db/schema.rb]: yeah acts_as_taggable is a bit crappy

File: app/controllers/images_controller.rb (hunk 2 above)
Comment (tlconnor) [inline app/controllers/images_controller.rb]: Fixed as discussed

File: test/models/image_test.rb (hunk 4 above)
Comment (tlconnor) [inline test/models/image_test.rb]: Damn good point, now fixed.

File: db/schema.rb (hunk 3 above)
Comment (bboe) [inline db/schema.rb]: Can you provide examples where code that you are using requires tag_id to be nullable? There are a number of fields that I was able to set to `null: false` on my application without issue.

**Comment by bboe** [comment]
image

Added a few comments. Is there a reason why this was a merge and not a rebase?

**Pull request merged**